### PR TITLE
Gps in ba

### PR DESCRIPTION
**Pull request by zpillio**
This modification uses GPS priors in BA. To make it work reconstruction normaliztion is disabled and alignment is applied before BA.

### CODE REVIEW

File: src/sfm/incremental_mapper.cc [hunk 1]
```diff
@@ -708,7 +718,7 @@ bool IncrementalMapper::AdjustGlobalBundle(
 
   // Normalize scene for numerical stability and
   // to avoid large scale changes in viewer.
-  reconstruction_->Normalize();
+  //reconstruction_->Normalize();
```
Comment (DeinFreund) [inline src/sfm/incremental_mapper.cc]: Just remove it completely.

File: src/sfm/incremental_mapper.cc [hunk 2]
```diff
@@ -598,6 +598,8 @@ IncrementalMapper::AdjustLocalBundle(
         ba_config.SetConstantCamera(camera_id_and_num_images_pair.first);
       }
     }
+/*
+    Using GPS priors avoid drift. TODO(zp) make using GPS prior optional.
 
     // Fix 7 DOF to avoid scale/rotation/translation drift in bundle adjustment.
```
[the remark this answers is not in the document]
Comment (DeinFreund) [reply src/sfm/incremental_mapper.cc]: Nvm this only works if we use the world coordinates for the camera positions.

File: src/sfm/incremental_mapper.cc [hunk 3]
```diff
@@ -598,6 +598,8 @@ IncrementalMapper::AdjustLocalBundle(
         ba_config.SetConstantCamera(camera_id_and_num_images_pair.first);
       }
     }
+/*
+    Using GPS priors avoid drift. TODO(zp) make using GPS prior optional.
```
[the remark this answers is not in the document]
Comment (zpillio) [reply src/sfm/incremental_mapper.cc]: Added a BA option

File: src/sfm/incremental_mapper.cc (hunk 2 above)
[the remark this answers is not in the document]
Comment (zpillio) [reply src/sfm/incremental_mapper.cc]: For that I think we need your modification in no_tvec_opt

[the remark this answers is not in the document]
Comment (DeinFreund) [reply src/sfm/incremental_mapper.cc]: Yes

**Pull request merged**